skills/quantum-vault/docs/ARCHITECTURE.md

```
@@ -0,0 +1,80 @@
+```cairo
+enum QuantumVaultStatus {
+    Unlocked,   // Initial state
+    Locked,     // Funds are locked
+    Released,   // Funds released successfully
+    Cancelled,  // Lock cancelled by owner
+}
+```
+
+## Data Flow
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)